core/src/main/kotlin/org/jetbrains/research/testspark/core/generation/llm/prompt/PromptBuilder.kt

```diff
@@ -106,8 +106,11 @@ internal class PromptBuilder(private var prompt: String) {
     ) = apply {
         val keyword = "\$${PromptKeyword.POLYMORPHISM.text}"
         if (isPromptValid(PromptKeyword.POLYMORPHISM, prompt)) {
-            var fullText = ""
-
+            // If polymorphismRelations is not empty, we add an instruction to avoid mocking classes if an instantiation of a sub-class is applicable
+            var fullText = when {
+                polymorphismRelations.isNotEmpty() -> "Use the following polymorphic relationships of classes present in the project. Use them for instantiation when necessary. Do not mock classes if an instantiation of a sub-class is applicable"
```

Comment:
I remember some prompt parts were stored in a file with all the string literals (and then accessed as `TestSparkBundle.getString("variable-name-of-string")`. Do we still do it or having a prompt part within the source code is fine?

Reply:
we are doing it everwhere yet, I suggest to create an issue and fix it in a separate PR